libxmf/XmfMuxer.cpp

```diff
@@ -0,0 +1,497 @@
+					!CopyVideoProjection(*parser_projection, &muxer_projection)) {
+					printf("\n Unable to copy video projection.\n");
+					return EXIT_FAILURE;
+				}
+				// Override the values that came from parser if set on command line.
+				if (projection_type != mkvparser::Projection::kTypeNotPresent) {
+					muxer_projection.set_type(
+						static_cast<MuxerProjType>(projection_type));
+					if (projection_type == mkvparser::Projection::kRectangular &&
+						projection_file != NULL) {
+						printf("\n Rectangular projection must not have private data.\n");
+						return EXIT_FAILURE;
+					}
+					else if ((projection_type == mkvparser::Projection::kCubeMap ||
+						projection_type == mkvparser::Projection::kMesh) &&
+						projection_file == NULL) {
+						printf("\n Mesh or CubeMap projection must have private data.\n");
+						return EXIT_FAILURE;
+					}
+					if (projection_file != NULL) {
+						std::string contents;
+						if (!libwebm::GetFileContents(projection_file, &contents) ||
+							contents.size() == 0) {
+							printf("\n Failed to read file \"%s\" or file is empty\n",
+								projection_file);
+							return EXIT_FAILURE;
+						}
+						if (!muxer_projection.SetProjectionPrivate(
+							reinterpret_cast<uint8_t*>(&contents[0]),
+							contents.size())) {
+							printf("\n Failed to SetProjectionPrivate of length %zu.\n",
+								contents.size());
+							return EXIT_FAILURE;
+						}
+					}
+				}
+				const float kValueNotPresent = mkvparser::Projection::kValueNotPresent;
+				if (projection_pose_yaw != kValueNotPresent)
+					muxer_projection.set_pose_yaw(projection_pose_yaw);
+				if (projection_pose_pitch != kValueNotPresent)
+					muxer_projection.set_pose_pitch(projection_pose_pitch);
+				if (projection_pose_roll != kValueNotPresent)
+					muxer_projection.set_pose_roll(projection_pose_roll);
+
+				if (!video->SetProjection(muxer_projection))
+					return EXIT_FAILURE;
+			}
+
+			if (track_name)
+				video->set_name(track_name);
+
+			video->set_codec_id(pVideoTrack->GetCodecId());
+
+			if (display_width > 0)
+				video->set_display_width(display_width);
+			if (display_height > 0)
+				video->set_display_height(display_height);
+			if (pixel_width > 0)
+				video->set_pixel_width(pixel_width);
+			if (pixel_height > 0)
+				video->set_pixel_height(pixel_height);
+			if (stereo_mode > 0)
+				video->SetStereoMode(stereo_mode);
+
+			const double rate = pVideoTrack->GetFrameRate();
+			if (rate > 0.0) {
+				video->set_frame_rate(rate);
+			}
+
+			size_t parser_private_size;
+			const unsigned char* const parser_private_data =
+				pVideoTrack->GetCodecPrivate(parser_private_size);
+
+			if (!strcmp(video->codec_id(), mkvmuxer::Tracks::kAv1CodecId)) {
+				if (parser_private_data == NULL || parser_private_size == 0) {
+					printf("AV1 input track has no CodecPrivate. %s is invalid.", inputPath);
+					return EXIT_FAILURE;
+				}
+			}
+
+			if (!strcmp(video->codec_id(), mkvmuxer::Tracks::kVp9CodecId) &&
```

Comment:
I guess this is only used if the input track uses VP9? we're still mostly using VP8, which is more common and less troublesome. Will the Remux function handle both VP8 and VP9 properly?

Reply:
This handles both VP8 and VP9 but advanced settings for vp9_profile and vp9_level is not enabled at this point (which I don't understand exactly what they are at this point). This could be added in future patches.